sdk/Sdk.Generators/FunctionsUtil.cs

```diff
@@ -28,19 +29,45 @@ internal static class FunctionsUtil
                 return false;
             }
 
-            foreach (var attr in methodSymbol.GetAttributes())
+            if (IsFunctionSymbol(methodSymbol, compilation))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        internal static bool IsFunctionSymbol(ISymbol symbol, Compilation compilation)
+        {
+            foreach (var attr in symbol.GetAttributes())
             {
                 if (attr.AttributeClass != null &&
                    SymbolEqualityComparer.Default.Equals(attr.AttributeClass, compilation.GetTypeByMetadataName(Constants.Types.FunctionName)))
                 {
-                    functionName = (string)attr.ConstructorArguments.First().Value!; // If this is a function attribute this won't be null
                     return true;
                 }
             }
 
             return false;
         }
 
+        internal static bool TryGetFunctionName(ISymbol symbol, Compilation compilation, out string? functionName) 
+        {
+            functionName = null;
+
+            var functionAttribute = symbol.GetAttributes()
+                .Where(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, compilation.GetTypeByMetadataName(Constants.Types.FunctionName)))
+                .FirstOrDefault();
```

Comment:
you can replace the Where with `FirstOrDefault` and remove the final call to `FirstOrDefault` at line 60